optuna/samplers/_tpe/sampler.py

```diff
@@ -396,15 +396,21 @@ def _sample_relative(
         indices_below, indices_above = _split_observation_pairs(scores, self._gamma(n), violations)
         # `None` items are intentionally converted to `nan` and then filtered out.
         # For `nan` conversion, the dtype must be float.
+        # In the call of `sample_relative`, this logic makes sense because we only have the
+        # intersection search space or group decomposed search space. This means one parameter
+        # misses the one trial, then the other parameter must miss the trial, in this call of
+        # `sample_relative`.
```

Comment:
This comment seems to have been added in #2688 to indicate that `list(config_values.values())[0]` is the correct logic.

But honestly, I don't understand what "This means one parameter misses the one trial, then the other parameter must miss the trial" means. Can you please tell me what kind of search space would include True in the param_mask? More specifically, what objective function would cause the [this condition](https://github.com/optuna/optuna/blob/master/optuna/samplers/_tpe/sampler.py#L669-L673) to be false?

Reply:
Thanks for the comment. I investigated that issue and changed the previous behavior by #4079 to make it consistent with `multivariate=False`.

Now, that should happen in a dynamic search space when `group=True`.